controllers/etcd_copy_backups_controller.go

```diff
@@ -0,0 +1,409 @@
+			}
+			return ctrl.Result{}, err
+		}
+
+		values, err := r.getValuesMap(etcdCopyBackupsTask)
+		if err != nil {
+			return ctrl.Result{}, err
+
+		}
+		if job, err = r.runCopyJob(ctx, etcdCopyBackupsTask, values, logger); err != nil {
+			if statusError := r.updateEtcdCopyTaskStatusLastError(ctx, etcdCopyBackupsTask, err); statusError != nil {
+				return ctrl.Result{}, fmt.Errorf("%s; %s", err, statusError)
+			}
+			return ctrl.Result{}, err
+
+		}
+		if err := r.addFinalizerToObject(ctx, job, logger); err != nil {
+			return ctrl.Result{}, err
+		}
+	} else if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := r.updateEtcdCopyBackupsTaskStatus(ctx, etcdCopyBackupsTask); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if etcdCopyBackupsTask.Status.LastOperation != nil &&
+		(reflect.DeepEqual(etcdCopyBackupsTask.Status.LastOperation.State, druidv1alpha1.LastOperationStateSucceeded) ||
+			reflect.DeepEqual(etcdCopyBackupsTask.Status.LastOperation.State, druidv1alpha1.LastOperationStateFailed)) {
+		return ctrl.Result{}, nil
+	}
+
+	return ctrl.Result{
+		RequeueAfter: 30 * time.Second,
+	}, nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) delete(ctx context.Context, etcdCopyBackupsTask *druidv1alpha1.EtcdCopyBackupsTask) (ctrl.Result, error) {
+	logger := r.logger.WithValues("task", kutil.Key(etcdCopyBackupsTask.Namespace, etcdCopyBackupsTask.Name).String(), "operation", "delete")
+	logger.Info("Starting operation")
+
+	if etcdCopyBackupsTask.Status.ObjectRef != nil {
+		if err := r.updateEtcdCopyBackupsTaskStatus(ctx, etcdCopyBackupsTask); err != nil {
+			return ctrl.Result{}, err
+		}
+		if etcdCopyBackupsTask.Status.LastOperation != nil && reflect.DeepEqual(etcdCopyBackupsTask.Status.LastOperation.State, druidv1alpha1.LastOperationStateActive) {
+			return ctrl.Result{
+				RequeueAfter: 30 * time.Second,
+			}, errors.New("The resource can not be deleted while still processing")
+		}
+
+		job := &batchv1.Job{}
+		if err := r.Get(ctx, kutil.Key(etcdCopyBackupsTask.Namespace, etcdCopyBackupsTask.Name+workerSuffix), job); client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{}, err
+		}
+		if err := r.removeFinalizerFromObject(ctx, job, logger); client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{}, err
+		}
+		if err := r.Delete(ctx, job); client.IgnoreNotFound(err) != nil {
+			return ctrl.Result{}, err
+		}
+	}
+	if err := r.removeFinalizerFromObject(ctx, etcdCopyBackupsTask, logger); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	logger.Info(fmt.Sprintf("Deleted ETCDCopyBackupTask %s/%s successfully.", etcdCopyBackupsTask.Namespace, etcdCopyBackupsTask.Name))
+	return ctrl.Result{}, nil
+}
+
+func (r *EtcdCopyBackupsTaskReconciler) updateEtcdCopyTaskStatusLastError(ctx context.Context, etcdCopyBackupsTask *druidv1alpha1.EtcdCopyBackupsTask, err error) error {
+	return kutil.TryUpdateStatus(ctx, retry.DefaultBackoff, r.Client, etcdCopyBackupsTask, func() error {
```

Comment:
Let's get rid of the try update logic and rather return an error. See https://github.com/gardener/gardener/blob/master/docs/development/kubernetes-clients.md#dont-retry-on-conflict.
I anyway don't expect conflicts today as there is likely just one controller updating the status.

Reply:
Also here, in addition to getting rid of the try update logic, shouldn't we rather update the status via a `Patch`?